x-pack/plugins/infra/public/pages/metrics/hosts/components/host_details_flyout/metadata/add_metadata_filter_button.tsx

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiToolTip, EuiButtonIcon } from '@elastic/eui';
+import { Filter, PhraseFilter, PhrasesFilter, ScriptedPhraseFilter } from '@kbn/es-query';
+import { buildMetadataFilter } from './build_metadata_filter';
+import { useMetricsDataViewContext } from '../../../hooks/use_data_view';
+import { useUnifiedSearchContext } from '../../../hooks/use_unified_search';
+import { useKibanaContextForPlugin } from '../../../../../../hooks/use_kibana';
+
+interface AddMetadataFilterButtonProps {
+  item: {
+    name: string;
+    value: string | string[] | undefined;
+  };
+}
+
+const filterAddedToastTitle = i18n.translate(
+  'xpack.infra.hostsViewPage.flyout.metadata.filterAdded',
+  {
+    defaultMessage: 'Filter was added',
+  }
+);
+
+export const AddMetadataFilterButton = ({ item }: AddMetadataFilterButtonProps) => {
+  const { dataView } = useMetricsDataViewContext();
+  const { searchCriteria } = useUnifiedSearchContext();
+  const [existingFilter, setExistingFilter] = useState<
+    PhrasesFilter | PhraseFilter | ScriptedPhraseFilter | Filter | undefined
+  >(() => undefined);
+  const {
+    services: {
+      data: {
+        query: { filterManager: filterManagerService },
+      },
+      notifications: { toasts: toastsService },
+    },
+  } = useKibanaContextForPlugin();
+
+  useMemo(() => {
+    const addedFilter = searchCriteria.filters.find((filter) => filter.meta.key === item.name);
+    if (addedFilter) {
+      setExistingFilter(addedFilter);
+    }
+  }, [item.name, searchCriteria.filters]);
```

Comment:
**suggestion**: `existingFilter` is derived from the `searchCriteria.filters` array and is not set anywhere else except here. In this scenario you don't need to set a state and create it, as you can directly memoize the derived value and use it:
```ts
const existingFilter = useMemo(
  () => searchCriteria.filters.find((filter) => filter.meta.key === item.name), 
  [item.name, searchCriteria.filters]
);
```

Reply:
Nice, as I mentioned in the other comment good idea- the state variable is not needed, updated 🙂